### ISSUE TITLE
Cobbler v3.3.0 does not honer the "auto_migrate_settings" setting

### ISSUE THREAD

**Issue by hveini** [cobbler/cobbler#2820]
If I change the "auto_migrate_settings" to "false", and restart cobblerd, the settings.yaml 
is still overwritten and "auto_migrate_settings" changes to "true":
New settings.yaml file is always generated (and old one backed up) when "cobblerd" service restarts, no matter what is the value of "auto_migrate_settings".

Way to reproduce (removed yum and build-and-install-rpms.sh -output):

```bash
[root@localhost cobbler]# git show
commit d8f60bbf14a838c8c8a1dba98086b223e35fe70a (HEAD -> v3.3.0, tag: v3.3.0, origin/release33)
Merge: 36c2ba14 452e4714
Author: Enno G <matrixfueller@gmail.com>
Date:   Mon Sep 20 16:55:41 2021 +0200

    Merge pull request #2794 from cobbler/fix/file-rce
    
    Fix multiple security vulnerabilites

[root@localhost cobbler]# ./docker/rpms/build-and-install-rpms.sh el8 docker/rpms/CentOS_8/CentOS8.dockerfile
[root@localhost cobbler]# yum install rpm-build/cobbler-3.3.0-1.el8.noarch.rpm
[root@localhost cobbler]# cd /etc/cobbler/
[root@localhost cobbler]# sed -i 's/auto_migrate_settings:.*/auto_migrate_settings: false/' settings.yaml 
[root@localhost cobbler]# grep auto_migrate_settings settings.yaml 
auto_migrate_settings: false
[root@localhost cobbler]# ls -dl settings*
drwxr-xr-x. 2 root root   119 10.10. 18:28 settings.d
-rw-r--r--. 1 root root 22389 10.10. 18.33 settings.yaml
[root@localhost cobbler]# systemctl restart cobblerd
[root@localhost cobbler]# grep auto_migrate_settings settings.yaml 
auto_migrate_settings: true
[root@localhost cobbler]# ls -l settings*
-rw-r--r--  1 root root 22389 10.10. 18:33 settings_20211010_18-33-50.yaml
drwxr-xr-x. 2 root root   119 10.10. 18:28 settings.d
-rw-r--r--. 1 root root  4844 10.10. 18:33 settings.yaml
[root@localhost cobbler]# cobbler --version
Cobbler 3.3.0
  source: ?, ?
  build time: Sun Oct 10 15:26:12 2021
[root@localhost cobbler]# cat /etc/redhat-release 
Rocky Linux release 8.4 (Green Obsidian)
```

**Comment by SchoolGuy** [comment]
Okay the root cause of this is a design error in the system how I imagined it:

Currently we pass as a default value `True`, so we do migrate per default. So far this is fine. The problem is in that chain that the value from the `settings.yaml` file is read but not taken into consideration. Thus the value from the daemon is used as the only source of truth. Since we did add this key to the settings, we do want to respect the value from the settings.

Workaround: Use the CLI argument to the `cobblerd` executable to disable the settings auto-migration.
